packages/babel-plugin-proposal-decorators/test/fixtures/2023-11-misc--to-es2015/.outer-class-binding-mutated/input.js

```diff
@@ -0,0 +1,145 @@
+{
+  "class binding in plain class, decorated field, and computed keys"
+  const errs = [];
+  const fns = [];
+  const capture = function (fn) {
+    fns.push(fn);
+    return () => {}
+  }
+  const assertUninitialized = function (fn) {
+    try {
+      fn();
+    } catch (err) {
+      errs.push(err);
+    } finally {
+      return () => {}
+    }
+  }
+
+  capture(() => K);
+  assertUninitialized(() => K);
+
+  class K {
+    @capture(() => K) @assertUninitialized(() => K) [(capture(() => K), assertUninitialized(() => K))]
+  }
+
+  const E = ReferenceError;
+  expect(errs.map(e => e.constructor)).toEqual([E, E, E]);
+
+  const C = K;
+  expect(fns.map(fn => fn())).toEqual([C, C, C]);
+
+  K = null;
+
+  expect(fns.map(fn => fn())).toEqual([null, C, C]);
+}
+
+{
+  "class binding in decorated class, decorated field, and computed keys"
+  const errs = [];
+  const fns = [];
+  const capture = function (fn) {
+    fns.push(fn);
+    return () => {}
+  }
+  const assertUninitialized = function (fn) {
+    try {
+      fn();
+    } catch (err) {
+      errs.push(err);
+    } finally {
+      return () => {}
+    }
+  }
+
+  @capture(() => K)
+  @assertUninitialized(() => K)
+  class K {
+    //todo: add the assertUninitialized decorator when we properly implement class tdz
```

Comment:
Could you for now assert that we throw the TDZ error, so that when we fix it we don't forget to update the test?